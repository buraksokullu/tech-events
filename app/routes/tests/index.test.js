import React from 'react';
import { shallow } from 'enzyme';
import RootRouter from '../index';

describe('RootRouter', () => {
  it('Should render correctly', () => {
    const component = shallow(<RootRouter />);
    expect(component).toMatchSnapshot();
  });

  it('renderHomeComponent should render correctly', () => {
    const component = shallow(<RootRouter />);
    const instance = component.instance();
    const dumProps = {
      dumValue: 'dumValue'
    };
    expect(instance.renderHomeComponent(dumProps)).toMatchSnapshot();
  });

  it('renderCampaignContentsComponent should render correctly', () => {
    const component = shallow(<RootRouter />);
    const instance = component.instance();
    const dumProps = {
      dumValue: 'dumValue'
    };
    expect(instance.renderCampaignContentsComponent(dumProps)).toMatchSnapshot();
  });

  it('redirectDashboardComponent should render correctly', () => {
    const component = shallow(<RootRouter />);
    const instance = component.instance();
    const dumProps = {
      dumValue: 'dumValue'
    };
    expect(instance.redirectDashboardComponent(dumProps)).toMatchSnapshot();
  });
});
