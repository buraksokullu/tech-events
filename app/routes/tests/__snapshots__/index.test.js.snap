// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootRouter Should render correctly 1`] = `
<Suspense
  fallback={<div />}
>
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/campaign-contents"
      render={[Function]}
    />
    <Route
      render={[Function]}
    />
  </Switch>
</Suspense>
`;

exports[`RootRouter redirectDashboardComponent should render correctly 1`] = `
<Redirect
  to="/"
/>
`;

exports[`RootRouter renderCampaignContentsComponent should render correctly 1`] = `
<UNDEFINED>
  <UNDEFINED
    dumValue="dumValue"
  />
</UNDEFINED>
`;

exports[`RootRouter renderDashboardComponent should render correctly 1`] = `
<UNDEFINED>
  <Dashboard
    dumValue="dumValue"
  />
</UNDEFINED>
`;
